src/Http/Http.Features/src/CookieOptions.cs

```diff
@@ -14,6 +16,25 @@ public class CookieOptions
     public CookieOptions()
     {
         Path = "/";
+        Extensions = new List<string>();
+    }
+
+    /// <summary>
+    /// Creates a copy of the given <see cref="CookieOptions"/>.
+    /// </summary>
+    public CookieOptions(CookieOptions options)
```

Comment:
We should validate we copy all properties:
See https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Mvc/Mvc.Core/test/ApplicationModels/PropertyModelTest.cs#L12
and
https://github.com/dotnet/aspnetcore/blob/9a74d3b4d0d56a0dca195b80cc6d0539308f7089/src/SignalR/clients/csharp/Client/test/UnitTests/HttpConnectionFactoryTests.cs#L77
for examples. Mostly just want it to fail if we add new properties.